pyterrier/terrier/_index.py

```diff
@@ -0,0 +1,81 @@
+import os
+from typing import List, Dict
+import pyterrier as pt
+
+
+class TerrierIndex(pt.Artifact):
+    """A Terrier index."""
+
+    ARTIFACT_TYPE = 'sparse_index'
+    ARTIFACT_FORMAT = 'terrier'
+    ARTIFACT_PACKAGE_HINT = 'python-terrier'
+
+    def __init__(self, path):
+        """Initialises a TerrierIndex for the given path."""
+        super().__init__(path)
+        self._index_ref = None
+        self._index_obj = None
+
+    def retriever(
+        self,
+        *,
+        controls: Dict = None,
+        properties: Dict = None,
+        metadata: List[str] = ["docno"],
+        num_results: int = 1000,
+        wmodel: str = 'DPH',
+        threads: int = 1,
+    ) -> pt.Transformer:
+        """Creates a ``pt.terrier.Retriever`` object for this index.
+
+        Args:
+            controls: The controls to set for this retriever. Controls are specific settings for a given search request.
+            properties: The properties to use for this retriever. Properties are settings that apply globally to the index.
+            metadata: The metadata fields to return for each search result.
+            num_results: The maximum number of results to return per query.
+            wmodel: The weighting model to use for scoring.
+            threads: The number of threads to use during retrieval.
+
+        Returns:
+            A retriever object for this index.
+        """
+        return pt.terrier.Retriever(self.index_obj(), controls, properties, metadata, num_results, wmodel)
+
+    def bm25(
```

Comment:
we probably need a discussion about which wmodels we will choose to expose.
also metadata?

Reply:
I'd rather keep this basic for now -- we can add all the bells and whistles in a separate PR to make sure this one remains manageable.